src/main/proto/uri.proto

```diff
@@ -20,20 +20,94 @@
+// of the Packet, for example in a CloudEvent Packet. <br>
+// UUri is used as a method to uniquely identify devices, services, and resources
+// on the  network.<br>
+// Where software is deployed, what the service is called along with a version and
+// the resources in the service. Defining a common URI for the system allows 
+// applications and/or services to publish and discover each other.<br>
+message UUri {
+  UAuthority authority = 1;
+  UEntity entity = 2;
+  UResource resource = 3;
+  
+  // Any additional details needed
+  repeated google.protobuf.Any details = 10;
+}
+
+
+// An Authority represents the deployment location of a specific Software Entity.
+// Data representation of an <b>Authority</b>.<br> An Authority consists of a device,
+// a domain and an address for the device.<br>
+// Device and domain names are used as part of the URI for device and service discovery.
+message UAuthority {
+  bool local = 1;     // True if local authority, false if remote authority
+  
+  // Remote authority information
+  oneof remote {
+    string name = 2;		// domain & device name as a string
+    bytes ip = 3; 			// IPv4 or IPv6 Address
+    bytes id = 4;			  // Unique ID for the device, could be a VIN, SHA 128, or any other identifier
+                        // *NOTE:* MAX length is 255 bytes
+  }
+}
+
+
+// Data representation of an <b> Software Entity - uE</b><br>
+// Software entities are distinguished by using a unique name and id 
+// along with the specific version of the software.
+// An Software Entity is a piece of software deployed somewhere on a uDevice.<br>
+// The Software Entity is used in the source and sink parts of communicating software.<br>
+// A uE that publishes events is a <b>Service</b> role.<br>
+// A uE that consumes events is an <b>Application</b> role.
+message UEntity {
+  string name = 1;    // Name of the entity
+  uint32 id = 2;      // The numeric ID for the uEntity assigned from the name/number registry
+  optional string version = 3; // optional version of the uEntity
+}
+
+
+// A service API - defined in the {@link UEntity} - has Resources and Methods. 
+// Both of these are represented by the UResource message.
+// A uResource represents a resource from a Service such as "door" and an optional
+// specific instance such as "front_left". In addition, it can optionally contain
+// the name of the resource Message type, such as "Door". The Message type matches
+// the protobuf service IDL that defines structured data types.
+// An UResource is something that can be manipulated/controlled/exposed by a service.
+// Resources are unique when prepended with UAuthority that represents the device and
+// UEntity that represents the service. 
+message UResource {
+  string name = 1;      // Name of the resource
+  string instance = 2;  // Instance of the resource
+  optional string message = 3;   // Message type for the resource
+  uint32 id = 4;        // ID that represents the method/topic for the resource
```

Comment:
why does the comment say "ID that represents the method/topic for the resource"?
I thought the ID represents the entire UResource.
Does it only represent part?

Reply:
fixed thanks